Index: hrtf_relearning/data/env config.txt
===================================================================
diff --git a/hrtf_relearning/data/env config.txt b/hrtf_relearning/data/env config.txt
deleted file mode 100644
--- a/hrtf_relearning/data/env config.txt	(revision a5f0f2be9bbb483146dbd9fe5f5e7afe0aad62c2)
+++ /dev/null	(revision a5f0f2be9bbb483146dbd9fe5f5e7afe0aad62c2)
@@ -1,14 +0,0 @@
-pycharm 2025.2
-python 3.11.9 virtual environment (-add git/bin to PATH variables)
-
-
-pip install git+https://github.com/pfriedrich-hub/slab.git
- git+https://github.com/pfriedrich-hub/pybinsim_tuil.git
- h5py h5netcdf metawear pyqt5 pynput pyfar
-
-
- optional:
-    -for cuda support (>=RTX 30xx):
-        pip install torch torchvision --index-url https://download.pytorch.org/whl/cu128
-        check with: torch.cuda.is_available()
- pip install git+https://github.com/pfriedrich-hub/freefield.git
\ No newline at end of file
